### PR TITLE
refactor predictor

### PR DESCRIPTION
**Pull request by zhangxiaoli73**
## What changes were proposed in this pull request?

1. add batchsize parameter for predict
2. add parameter repeatMemory to appoint whether clone the predict results

## How was this patch tested?
unit test

### CODE REVIEW

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/abstractnn/AbstractModule.scala [hunk 1]
```diff
@@ -470,17 +470,24 @@ abstract class AbstractModule[A <: Activity: ClassTag, B <: Activity: ClassTag,
...
    * @param dataset dataset for prediction
+   * @param batchSize total batchSize for all partitions.
+   *                  if -1, default is 4 * partitionNumber of datatset
+   * @param repeatMemory whether to use same memory for each batch predict results
    */
-  def predict(dataset: RDD[Sample[T]]): RDD[Activity] = {
-    Predictor(this).predict(dataset)
+  def predict(dataset: RDD[Sample[T]],
+              batchSize: Int = -1,
+              repeatMemory: Boolean = false): RDD[Activity] = {
```
Comment (jason-dai) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/abstractnn/AbstractModule.scala]: why false by default?

Comment (zhangxiaoli73) [reply spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/abstractnn/AbstractModule.scala]: In our user group, a user uses rdd.take() to get repeated predict results. And he thinks it is confused for him. Other users may have such confuse too. So I use default "false" to get no repeat results.

Comment (jason-dai) [reply spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/abstractnn/AbstractModule.scala]: ok - change the name to sth like "shareBuffer"

Comment (zhangxiaoli73) [reply spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/abstractnn/AbstractModule.scala]: ok

**Comment by zhangxiaoli73** [comment]
jekins pass

**Pull request merged**